MSAL/test/unit/native_auth/public/MSALNativeAuthUserAccountResultTests.swift

```diff
@@ -46,30 +87,55 @@ class MSALNativeAuthUserAccountResultTests: XCTestCase {
 
         sut = MSALNativeAuthUserAccountResult(
             account: account!,
-            authTokens: MSALNativeAuthTokens(accessToken: accessToken, refreshToken: refreshToken, rawIdToken: rawIdToken),
+            rawIdToken: rawIdToken,
             configuration: MSALNativeAuthConfigStubs.configuration,
             cacheAccessor: cacheAccessorMock
         )
         try super.setUpWithError()
     }
 
-    // MARK: Call delegate properly tests
-
-    func test_whenAccountAndTokenExist_itReturnsCorrectData() {
-        let expectation = expectation(description: "CredentialsController")
-        let authTokens = MSALNativeAuthUserAccountResultStub.authTokens
-        let mockDelegate = CredentialsDelegateSpy(expectation: expectation, expectedResult: MSALNativeAuthTokenResult(authTokens: authTokens))
-        mockDelegate.expectedAccessToken = authTokens.accessToken.accessToken
-        mockDelegate.expectedExpiresOn = authTokens.accessToken.expiresOn
-        mockDelegate.expectedScopes = authTokens.accessToken.scopes.array as? [String] ?? []
-        sut.getAccessToken(delegate: mockDelegate)
-        wait(for: [expectation], timeout: 1)
-    }
-
     // MARK: - sign-out tests
 
     func test_signOut_successfullyCallsCacheAccessor() {
         sut.signOut()
         XCTAssertTrue(cacheAccessorMock.clearCacheWasCalled)
     }
+    
+    // MARK: - error tests
+    
+    func test_errorWithInnerError() {
+        let contextCorrelationId = UUID()
+        let context = MSALNativeAuthRequestContext(correlationId: contextCorrelationId)
+        
+        let result = sut.createRetrieveAccessTokenError(error: errorWithInnerErrorMock,
+                                                        context: context)
+        
+        XCTAssertEqual(result.errorDescription, "inner_user_info_error_description")
+        XCTAssertEqual(result.errorCodes, [])
+        XCTAssertEqual(result.correlationId.uuidString, innerCorrelationId)
+    }
+    
+    func test_errorWithoutInnerError() {
+        let contextCorrelationId = UUID()
+        let context = MSALNativeAuthRequestContext(correlationId: contextCorrelationId)
+        
+        let result = sut.createRetrieveAccessTokenError(error: errorWithoutInnerErrorMock,
+                                                        context: context)
+        
+        XCTAssertEqual(result.errorDescription, "user_info_error_description")
+        XCTAssertEqual(result.errorCodes, [-3003])
+        XCTAssertEqual(result.correlationId.uuidString, withoutInnerCorrelationId)
```

Comment:
There's something I don't understand from here. The context is created in line 120.
Shouldn't the assert in the line 127 be equal to the context of line 120?

Reply:
We get the correlation id from the error. If not possible, we get it from the context. Correct me if I'm wrong @nilo-ms.

Reply:
I'm taking a look at the implementation and I've written a [comment](https://github.com/AzureAD/microsoft-authentication-library-for-objc/pull/2159/files#r1622184786) related to this one there, so I'll resolve this to avoid duplication.